src/client/datascience/jupyter/interpreter/jupyterInterpreterService.ts

```diff
@@ -59,15 +59,15 @@ export class JupyterInterpreterService {
     // To be run one initial time. Check our saved locations and then current interpreter to try to start off
     // with a valid jupyter interpreter
     public async setInitialInterpreter(token?: CancellationToken): Promise<PythonEnvironment | undefined> {
-        if (!this.getInitialInterpreterPromise) {
-            this.getInitialInterpreterPromise = this.getInitialInterpreterImpl(token).then((result) => {
-                // Set ourselves as a valid interpreter if we found something
-                if (result) {
-                    this.changeSelectedInterpreterProperty(result);
-                }
-                return result;
-            });
-        }
+        // if (!this.getInitialInterpreterPromise) {
+        this.getInitialInterpreterPromise = this.getInitialInterpreterImpl(token).then((result) => {
+            // Set ourselves as a valid interpreter if we found something
+            if (result) {
+                this.changeSelectedInterpreterProperty(result);
```

Comment:
With the new change, this method will get executed everytime, previously we'd cache the result.
I don't think thats the desired effect.

Reply:
Caching the result was the problem, if you said don't install, then the export button won't ask again and automatically fail. When it should ask again.